### ISSUE TITLE
strange parsing of comments

### ISSUE THREAD

**Issue by steveklabnik** [sunng87/handlebars-rust#202]
I have this:

```hbs
                {{!--
                    <li><a href="{{up-dir nest-count}}{{base-url}}index.html">{{this.title}}</a></li>
                --}}
```

this prints out

```text
               index.html">Getting Started</a></li>
                --}}
                
```

in my output.

I *think* the issue is, `nest-count` has a `-`, `base-url` has a `-`, then `}}` comes, so it tries to end the comment. Not 100% sure though!

**Comment by sunng87** [comment]
I see. This looks weird. Let me add a test case to reproduce this.

**Comment by sunng87** [comment]
Checked the grammer, `{{!-- --}}` is not supported for now. 
I will add a quick implementation for that.

**Comment by sunng87** [comment]
Released in 0.31.0
